pkg/cloud/isolated_network.go

```diff
@@ -247,8 +247,9 @@ func (c *client) GetOrCreateIsolatedNetwork(
 		if err = c.CreateIsolatedNetwork(zone, isoNet); err != nil {
 			return errors.Wrap(err, "creating a new isolated network")
 		}
+	} else {
+		isoNet.Spec.ID = net.ID
 	}
-	isoNet.Spec.ID = net.ID
```

Comment:
why is this change being made? I think we want to set the ID if we're creating a new isolated network right?

Reply:
The following test fails without this change. 

https://github.com/kubernetes-sigs/cluster-api-provider-cloudstack/blob/b9dc417e24dbd90e1c1eb5573d5c735444eca64d/pkg/cloud/isolated_network_test.go#L218

We were skipping this test prior.f

I'm going to add some comments, but basically there's a back and forth between the CloudStackIsolatedNetwork CRD/type here and the Network type.

I would like the two to be the same thing, but haven't been able to resolve that just yet.

In this case, if a CloudStackIsolatedNetwork is passed to this method without the ID set, and the Isolated network exists, then we will actually fail to set the ID of that network on the CloudStackIsolatedNetwork.